docs/source/class_source.rst

```diff
@@ -0,0 +1,335 @@
+expect ``classes[0,0]`` to be ``input_src.images[0]``.
+
+No further class selection or order occurs during classification.
+Those methods are broken out into other components.
+
+Currently ASPIRE has a single classification algorithm known as
+``RIRClass2D``.  This algorithm uses multiple applications of PCA in
+conjunction with Bispectrum analysis to identify nearest neighbors in
+a rotationally invariant feature space.
+
+.. mermaid::
+
+   classDiagram
+      class Class2D{
+	  +classify()
+      }
+    Class2D <|-- RIRClass2D
+
+Class Selectors
+***************
+
+Class Selectors consume the output of ``Class2D`` and attempt to order
+and/or filter classes down to a selection.  Selecting the "best"
+classes in CryoEM problems is still an area of active research.  Some
+common methods are provided, along with an extensible base interface.
+
+Generally, Class Selection comes in two flavors depending on what
+information is required to perform the selection.
+
+Local Class Selectors
+---------------------
+
+For "Local" class selection, we will attempt to use only the
+information returned from ``Class2D``.  In the case of ``RIRClass2D``
+this would primarily be a network of ``distances`` as measured in the
+compressed feature space.
+
+This approach has two main advantages.  First, we already have this
+information computed as part of classification.  Second, it allows us
+to register and stack a relatively small subset of the "best" classes.
+Because registration and alignment are computationally expensive this
+can reduce pipeline run times by an order of magnitude.
+
+.. mermaid::
+
+   classDiagram
+      class ClassSelector{
+	 +select()
+	 }
+       ClassSelector <|-- TopClassSelector
+       ClassSelector <|-- RandomClassSelector
+       ClassSelector <|-- NeighborVarianceClassSelector
+       ClassSelector <|-- DistanceClassSelector
+       ClassSelector o-- GreedyClassRepulsion
+
+Global Class Selectors
+----------------------
+
+Global Class Selection techniques first compute the entire collection
+registered and aligned class averages, then compute some quality
+measure on all classes.
+
+These are most similar to the historical MATLAB approaches, sometimes
+referred to as "out-of-core" methods.  It is believed that many legacy
+MATLAB experiments computed contrast (variance) of all class averaged
+images, sorted the class averages to express highest contrast,
+potentially avoiding classes with views we've already seen.  This can
+be accomplished now by using the ``ContrastImageQualityFunction`` in a
+``GlobalWithRepulsionClassSelector``.
+
+An SNR based approach is also provided, and a Bandpass method should
+be implemented in a future release.  Again, these components are fully
+customizable and the base interfaces were designed with algorithm
+developers in mind.
+
+Implementing concrete ``GlobalClassSelector`` leverage subcomponents
+described below.
+
+.. mermaid::
```

Comment:
To break it up we'll lose the connections, the main reason to draw them... If this was another software, I would just be happy they actually had this and tap cmd-+ (ctrl-+) few times when on my laptop.

I think we may be able to change the style/css of our doc page easier than redoing all the uml stuff, but I haven't looked into that. FWIW I picked `mermaid` so we can have one tool that will work for both docs and gallery, but it will require other compromises (potentially like this).

I'm open to change, but the feedback I got from Josh is that these diagrams are important.

Reply:
Works for me.